src/Services/UserPasswords/UserPasswordsService.cs

```diff
@@ -0,0 +1,247 @@
+	{
+		private readonly AppDbContext _db;
+		private readonly ILogger<UserPasswordsService> _logger;
+		public UserPasswordsService(
+			AppDbContext db,
+			ILogger<UserPasswordsService> logger
+		)
+		{
+			_db = db;
+			_logger = logger;
+		}
+
+		public async Task<List<UserPasswordDto>> Get()
+		{
+			var entityList = await _db.UserPasswords.ToListAsync();
+
+			List<UserPasswordDto> usersDtoList = new List<UserPasswordDto>();
+
+			foreach (var entity in entityList)
+			{
+				usersDtoList.Add(new UserPasswordDto
+				{
+					Id = entity.Id,
+					Title = entity.Title,
+					Username = entity.Username,
+					CreateDate = entity.CreateDate
+				});
+			}
+
+			return usersDtoList;
+		}
+
+		public async Task<Result<UserPasswordDto>> Get(int id)
+		{
+			var entity = await _db.UserPasswords.FindAsync(id);
+
+			if (entity == null)
+			{
+				return new Result<UserPasswordDto>
+				{
+					IsSuccssfull = false,
+					Message = "User password not found",
+				};
+			}
+
+			return new Result<UserPasswordDto>
+			{
+				IsSuccssfull = true,
+				Data = new UserPasswordDto
+				{
+					Id = entity.Id,
+					Title = entity.Title,
+					Username = entity.Username,
+					CreateDate = entity.CreateDate
+				}
+			};
+		}
+
+		public async Task<Result<UserPasswordDto>> Insert(UserPasswordFormDto user)
+		{
+			var transaction = _db.Database.BeginTransaction();
+
+			try
+			{
+				UserPassword entity = new UserPassword();
+
+				entity.Title = user.Title;
+				entity.Username = user.Username;
+				entity.Password = user.Password;
+				entity.Site = user.Site;
+				entity.Category = user.Category;
+				entity.CreateDate = DateTime.Now;
+
+				_db.UserPasswords.Add(entity);
+
+				await _db.SaveChangesAsync();
+
+				_logger.LogInformation(
+					"Saved a new user password to database with title '{Title}' and id: {Id}",
+					entity.Title,
+					entity.Id
+				);
+
+				transaction.Commit();
+
+				return new Result<UserPasswordDto>
+				{
+					IsSuccssfull = true,
+					Message = "Inserted",
+					Data = new UserPasswordDto
+					{
+						Id = entity.Id,
+						Username = entity.Username,
+						Title = entity.Title,
+						CreateDate = entity.CreateDate
+					}
+				};
+			}
+			catch (Exception ex)
+			{
+				_logger.LogError("Failed to save user password because of exception: '{Message}'", ex.Message);
+
+				transaction.Rollback();
+
+				return new Result<UserPasswordDto>
+				{
+					IsSuccssfull = false,
+					Message = "Error while trying to save user password due to technical reason with code: " + ex.HResult,
+				};
+			}
+		}
+
+		public async Task<Result<UserPasswordDto>> Update(int id, UserPasswordFormDto user)
+		{
+			var transaction = _db.Database.BeginTransaction();
+
+			try
+			{
+				var entity = await _db.UserPasswords.FindAsync(id);
+
+				if (entity == null)
+				{
+					return new Result<UserPasswordDto>
+					{
+						IsSuccssfull = false,
+						Message = "User password not found",
+						Data = new UserPasswordDto
```

Comment:
you cant fill data here is the entity is null so it would cause a null ref exception